libs/blocks/quiz/quiz.js

```diff
@@ -0,0 +1,334 @@
+  const quizSections = document.querySelector('.quiz-footer');
+  const a = createTag('a', { href: fragmentURL });
+  quizSections.append(a);
+  const { default: createFragment } = await import('../fragment/fragment.js');
+  await createFragment(a);
+}
+
+const App = ({
+  initialIsDataLoaded = false,
+  preQuestions = {}, initialStrings = {},
+}) => {
+  const [btnAnalytics, setBtnAnalytics] = useState(null);
+  const [isBtnClicked, setIsBtnClicked] = useState(false);
+  const [countSelectedCards, setCountOfSelectedCards] = useState(0);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [isDataLoaded, setDataLoaded] = useState(initialIsDataLoaded);
+  const [nextQuizViewsExist, setNextQuizViewsExist] = useState(true);
+  const [prevStepIndicator, setPrevStepIndicator] = useState([]);
+  const [questionData, setQuestionData] = useState(preQuestions.questionData || {});
+  const [questionList, setQuestionList] = useState(preQuestions.questionList || {});
+  const [selectedCards, setSelectedCards] = useState({});
+  const [selectedQuestion, setSelectedQuestion] = useState(preQuestions || null);
+  const [stringData, setStringData] = useState(initialStrings || {});
+  const [stringQList, setStringQList] = useState(preQuestions.stringQList || {});
+  const [totalSteps, setTotalSteps] = useState(3);
+  const initialUrlParams = getUrlParams();
+  const [urlParam, setUrlParam] = useState(initialUrlParams);
+  const [userSelection, updateUserSelection] = useState([]);
+  const [userFlow, setUserFlow] = useState([]);
+  const validQuestions = useMemo(() => [], []);
+  const knownParams = useMemo(() => ['martech', 'milolibs', 'quiz-data'], []);
+
+  useEffect(() => {
+    (async () => {
+      const [questions, dataStrings] = await getQuizData();
+      const qMap = {};
+      questions.questions.data.forEach((question) => {
+        qMap[question.questions] = question;
+        validQuestions.push(question.questions);
+      });
+
+      const strMap = {};
+      dataStrings.questions.data.forEach((question) => {
+        strMap[question.q] = question;
+      });
+
+      setUserFlow([questions.questions.data[0].questions]);
+
+      setStringData(dataStrings);
+      setQuestionData(questions);
+      setStringQList(strMap);
+      setQuestionList(qMap);
+      setDataLoaded(true);
+      document.body.classList.add('quiz-page');
+    })();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setQuestionData, setStringData, setStringQList, setQuestionList]);
+
+  useEffect(() => {
+    function handlePopState() {
+      window.location.reload();
+    }
+    if (isDataLoaded) {
+      window.addEventListener('popstate', handlePopState);
+      return () => {
+        window.removeEventListener('popstate', handlePopState);
+      };
+    }
+    return () => {};
+  }, [knownParams, isDataLoaded]);
+
+  useEffect(() => {
+    if (userFlow.length) {
+      const currentFlow = userFlow.shift();
+      if (!currentFlow?.length) {
```

Comment:
Do you need to check if there is a length if currentFlow is null?

Reply:
I updated the code to test both userFlow and currentFlow in their respective locations before looking within them.